src/components/Login.tsx

```diff
@@ -35,9 +38,67 @@ const Login = (props: Props) => {
     setPassword(value);
   }
 
+  const onFirstNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const { value } = event.target;
+    setFirstName(value);
+  }
+
+  const onLastNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const { value } = event.target;
+    setLastName(value);
+  }
+
   const generateAgent = async (event: React.SyntheticEvent) => {
     setLoading(true);
     let agentStatus = await ad4mClient.agent.generate(password);
+    const agentPerspective = await ad4mClient.perspective.add(
+      "My perspective"
+    );
+    const links = [];
+
+    if (firstName) {
+      const link = await ad4mClient.perspective.addLink(
+        agentPerspective.uuid,
+        new Link({
+          source: 'ad4m://profile',
```

Comment:
Can we make it a constant in a new file like `triples`, and other constant strings used?